edweiss-app/app/(app)/deck/[id]/card/index.tsx

```diff
@@ -0,0 +1,218 @@
+/**
+ * Create a card or edit an existing card screen
+ * 
+ * @file index.tsx
+ * @description Create a card or edit an existing card screen
+ * @author Tuan Dang Nguyen
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import RouteHeader from '@/components/core/header/RouteHeader';
+import { ApplicationRoute } from '@/constants/Component';
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import Icon from '@/components/core/Icon';
+import RichText from '@/components/core/rich-text/RichText';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { callFunction } from '@/config/firebase';
+import { iconSizes } from '@/constants/Sizes';
+import { useRepositoryDocument } from '@/hooks/repository';
+import { useStringParameters } from '@/hooks/routeParameters';
+import Memento from '@/model/memento';
+import { checkDupplication_EmptyField } from '@/utils/memento/utilsFunctions';
+import { router } from 'expo-router';
+import React, { useState } from 'react';
+import { Modal } from 'react-native';
+import { DecksRepository } from '../../_layout';
+
+// ------------------------------------------------------------
+// ----------------- CreateCardScreen Component ----------------
+// ------------------------------------------------------------
+
+/**
+ * Create or edit card screen
+ * User can create a new card or edit an existing card with a question and an answer
+ * 
+ * @returns {ApplicationRoute} Screen to create or edit a card
+ */
+const CreateEditCardScreen: ApplicationRoute = () => {
+    const { deckId, mode, prev_question, prev_answer, cardIndex } = useStringParameters();
```

Comment:
If those parameters are push from another page, think to use `useRouteParameters`instead

Reply:
Yup, I will try this new RouteParameters things. Funny because I have never known that it has been existing till now xDD

Reply:
Do you happen to know what is "ApplicationRouteSignature"?

Reply:
> Do you happen to know what is "ApplicationRouteSignature"?

This is the pattern of the parameters given by the push of the router. Take a look how I use it in Archive.tsx : 
```
export const ArchiveRouteSignature: ApplicationRouteSignature<{
    courseId: string,
    assignments: { id: string, data: AssignmentWithColor }[]
}> = {
    path: `/courses/[id]/archive`
}
```

Reply:
I see, thank you :D